pkg/controller/machinepool/alibabacloudactuator.go

```diff
@@ -123,3 +132,15 @@ func (a *AlibabaCloudActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment,
 
 	return installerMachineSets, true, nil
 }
+
+// Get the image ID from an existing master machine.
+func getAlibabaCloudImageID(masterMachine *machineapi.Machine, logger log.FieldLogger) (string, error) {
+	providerSpec, err := alibabacloudprovider.ProviderSpecFromRawExtension(masterMachine.Spec.ProviderSpec.Value)
+	if err != nil {
+		logger.WithError(err).Warn("cannot decode AlibabaCloudMachineProviderSpec from master machine")
+		return "", errors.Wrap(err, "cannot decode AlibabaCloudMachineProviderSpec from master machine")
+	}
+	imageID := providerSpec.ImageID
```

Comment:
We can probably expect `providerSpec.ImageID` to be set for the master machine but we should check that here and provide an error if not set similar to [getAWSAMIID()](https://github.com/openshift/hive/blob/fa6d7dfb0d468c45a5b9fb89795c927af3aaa7b5/pkg/controller/machinepool/awsactuator.go#L247-L250).

Reply:
fixed